stats.go

```diff
@@ -207,13 +233,74 @@ func (r *timerNoReporterSink) ReportTimer(name string, tags map[string]string, i
 	r.timer.unreported.values = append(r.timer.unreported.values, interval)
 	r.timer.unreported.Unlock()
 }
+func (r *timerNoReporterSink) ReportHistogramValue(name string, tags map[string]string, buckets Buckets, value float64) {
+}
+func (r *timerNoReporterSink) ReportHistogramDuration(name string, tags map[string]string, buckets Buckets, interval time.Duration) {
+}
 func (r *timerNoReporterSink) Capabilities() Capabilities {
-	return capabilitiesReportingTagging
+	return capabilitiesReportingTaggingNoHistograms
 }
 func (r *timerNoReporterSink) Flush() {
 }
 
-// NullStatsReporter is an implementatin of StatsReporter than simply does nothing.
+type histogram struct {
+	name            string
+	tags            map[string]string
+	reporter        StatsReporter
+	buckets         Buckets
+	cachedHistogram CachedHistogram
+}
+
+func newHistogram(
+	name string,
+	tags map[string]string,
+	r StatsReporter,
+	buckets Buckets,
+	cachedHistogram CachedHistogram,
+) *histogram {
+	return &histogram{
+		name:            name,
+		tags:            tags,
+		reporter:        r,
+		buckets:         buckets,
+		cachedHistogram: cachedHistogram,
+	}
+}
+
+func (h *histogram) RecordValue(value float64) {
+	if h.cachedHistogram != nil {
+		h.cachedHistogram.ReportHistogramValue(value)
```

Comment:
Histograms can be aggregated. Tally aggregates counters & gauges inside tally itself instead of the reporter.

It might make sense to aggregate in tally and to report/flush in a tick loop instead of reporting every single histogram.